cpp/modules/deck.gl/core/src/arrow/row.cc

```diff
@@ -41,7 +41,7 @@ auto Row::getInt(const std::string& columnName, int defaultValue) const -> int {
 
   auto chunk = this->_getChunk(columnName);
   auto doubleValue = this->_getDouble(chunk);
-  if (doubleValue.has_value()) {
+  if (doubleValue) {
```

Comment:
This seems like a JavaScript-ism to me. Maybe rename to `doubleOption`? The `value` part of the name combined with using `auto` makes me think `doubleValue` actually has type `double., for which this if statement doesn't really make sense.

Reply:
I also agree that using `has_value` is more verbose and improves the readability, especially when using `auto`. @ibgreen seems to prefer leaving that out, so I was trying to be consistent.

Updated `auto` to explicit type as a middle ground

Reply:
I can see both view points. pointers in C++ are routinely compared as boolean values. 

`if (pointer)` vs `if (pointer === nullptr)`

And C++ has the conversion to boolean built-in.

std::optional is clearly designed to mimic the pointer idiom. It support similar style if checking, and can even be "dereferenced" using the `*` operator (instead of `get_value`)

So - we may want to pick a style (e.g. pointer like vs explicit has/get_value calls) since we are increasingly making use of `std::optional`.